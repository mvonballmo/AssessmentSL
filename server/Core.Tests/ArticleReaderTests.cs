using System;
using System.Net.Http;
using System.Threading.Tasks;
using Microsoft.Extensions.DependencyInjection;
using NUnit.Framework;
using Snapshooter.NUnit;

namespace Core.Tests;

public class ArticleReaderTests
{
    [Test]
    public async Task TestGetArticles()
    {
        var provider = CreateProvider();
        var articleReader = provider.GetRequiredService<ArticleReader>();

        var articles = await articleReader.GetArticles();
        
        Snapshot.Match(articles);
    }

    private IServiceProvider CreateProvider()
    {
        var services = new ServiceCollection();

        services
            .AddSingleton<HttpClient>()
            .AddSingleton<ArticleReader>();

        return services.BuildServiceProvider();
    }
}