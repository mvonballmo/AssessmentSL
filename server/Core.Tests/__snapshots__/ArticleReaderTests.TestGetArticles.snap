[
  {
    "Id": 14044,
    "Title": "NOAA seeking information on commercial space situational awareness data",
    "ImageUrl": "https://spacenews.com/wp-content/uploads/2022/02/Screen-Shot-2022-02-11-at-11.07.23-AM.png",
    "Rating": 0,
    "Comments": []
  },
  {
    "Id": 14043,
    "Title": "SpaceX to launch “next-generation satellite-servicing vehicle” for Northrop Grumman",
    "ImageUrl": "https://www.teslarati.com/wp-content/uploads/2022/02/MEVMEP.4K4002-crop-c.jpg",
    "Rating": 0,
    "Comments": []
  },
  {
    "Id": 14042,
    "Title": "Lynk reveals mobile network contracts",
    "ImageUrl": "https://spacenews.com/wp-content/uploads/2022/02/rsz_lynk.jpg",
    "Rating": 0,
    "Comments": []
  },
  {
    "Id": 14040,
    "Title": "Washington state space economy more than doubles",
    "ImageUrl": "https://spacenews.com/wp-content/uploads/2020/03/Starlink-470x470.jpg",
    "Rating": 0,
    "Comments": []
  },
  {
    "Id": 14039,
    "Title": "U.S. and international partners to share launch cost of new communications satellite",
    "ImageUrl": "https://spacenews.com/wp-content/uploads/2020/10/WGS-11_BeautyShot_FINAL-copy-253x253.jpg",
    "Rating": 0,
    "Comments": []
  },
  {
    "Id": 14038,
    "Title": "U.S. and allies to strengthen cooperation in space",
    "ImageUrl": "https://spacenews.com/wp-content/uploads/2022/02/180928-D-XX000-228.jpg",
    "Rating": 0,
    "Comments": []
  },
  {
    "Id": 14037,
    "Title": "NASA to Provide Media Update on Artemis I Moon Mission",
    "ImageUrl": "https://www.nasa.gov/sites/default/files/thumbnails/image/ksc-20220119-ph-csh01_0016_large.jpeg?itok=iIJTdmca",
    "Rating": 0,
    "Comments": []
  },
  {
    "Id": 14035,
    "Title": "With eyes on reuse, Relativity plans rapid transition to Terran R engines",
    "ImageUrl": "https://cdn.arstechnica.net/wp-content/uploads/2022/02/E4B-Thumbnail-1-3.png",
    "Rating": 0,
    "Comments": []
  },
  {
    "Id": 14034,
    "Title": "CAPSTONE lunar cubesat mission to launch this spring",
    "ImageUrl": "https://spacenews.com/wp-content/uploads/2021/08/capstone.jpg",
    "Rating": 0,
    "Comments": []
  },
  {
    "Id": 14033,
    "Title": "SpaceX to send retired Starship to local Texas airport, says Elon Musk",
    "ImageUrl": "https://www.teslarati.com/wp-content/uploads/2022/02/Starbase-022022-NASASpaceflight-bocachicagal-S22-B5-SN15-SN16-4-crop-c.jpg",
    "Rating": 0,
    "Comments": []
  }
]
